src/test/java/io/supertokens/test/bulkimport/apis/BulkImportBackgroundJobManagerAPITest.java

```diff
@@ -0,0 +1,125 @@
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STARTED));
+        Main main = process.getProcess();
+
+        if (StorageLayer.getBaseStorage(main).getType() != STORAGE_TYPE.SQL || StorageLayer.isInMemDb(main)) {
+            return;
+        }
+
+        //by default inactive
+        {
+            JsonObject currentStatus = loadBulkImportCronjobStatus(main);
+
+            assertEquals("OK", currentStatus.get("status").getAsString());
+            assertEquals("INACTIVE", currentStatus.get("jobStatus").getAsString());
+        }
+
+        //starting it makes it active
+        {
+            JsonObject startResponse = startBulkImportCronjob(main, 100);
+            assertEquals("OK", startResponse.get("status").getAsString());
+            assertEquals("ACTIVE", startResponse.get("jobStatus").getAsString());
+
+            JsonObject currentStatus = loadBulkImportCronjobStatus(main);
+            assertEquals("OK", currentStatus.get("status").getAsString());
+            assertEquals("ACTIVE", currentStatus.get("jobStatus").getAsString());
+        }
+
+
+        //stopping it makes it inactive
+        {
+            JsonObject startResponse = stopBulkImportCronjob(main);
+            assertEquals("OK", startResponse.get("status").getAsString());
+            assertEquals("INACTIVE", startResponse.get("jobStatus").getAsString());
+
+            JsonObject currentStatus = loadBulkImportCronjobStatus(main);
+            assertEquals("OK", currentStatus.get("status").getAsString());
+            assertEquals("INACTIVE", currentStatus.get("jobStatus").getAsString());
+        }
+    }
+
+    private static JsonObject startBulkImportCronjob(Main main, int batchSize) throws HttpResponseException, IOException {
+        return sendCommandForBulkImportCronjob(main, "START", batchSize);
+    }
+
+    private static JsonObject stopBulkImportCronjob(Main main) throws HttpResponseException, IOException {
+        return sendCommandForBulkImportCronjob(main, "STOP", null);
+    }
+
+    private static JsonObject sendCommandForBulkImportCronjob(Main main, String command, Integer batchSize) throws HttpResponseException, IOException {
+        JsonObject request = new JsonObject();
+        request.addProperty("command", command);
+        if (batchSize != null) {
+            request.addProperty("batchSize", batchSize);
+        }
+        return HttpRequestForTesting.sendJsonPOSTRequest(main, "",
+                BACKGROUNDJOB_MANAGER_ENDPOINT,
+                request, 1000, 10000, null, Utils.getCdiVersionStringLatestForTests(), null);
+    }
+
+    private static JsonObject loadBulkImportCronjobStatus(Main main) throws HttpResponseException, IOException {
+        return HttpRequestForTesting.sendGETRequest(main, "", BACKGROUNDJOB_MANAGER_ENDPOINT, null,
+                1000, 1000, null, Utils.getCdiVersionStringLatestForTests(), null);
+    }
+
```

Comment:
What's the behaviour if the job is stopped when an import is happening? can we test that as well?

Reply:
Processing is stopped. Added a new test for that.